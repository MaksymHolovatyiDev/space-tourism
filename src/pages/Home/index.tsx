import {ExploreButton} from 'entities/explore-button';
import style from './styled.module.scss';
import {Typography} from 'shared/Typography';

function Home() {
  return (
    <main className={style.root}>
      <div className={style.container}>
        <Typography text="heading 5">So, you wat to travel to</Typography>
        <Typography text="heading 1">Space</Typography>
        <Typography>
          Let’s face it; if you want to go to space, you might as well genuinely
          go to outer space and not hover kind of on the edge of it. Well sit
          back, and relax because we’ll give you a truly out of this world
          experience!
        </Typography>
      </div>
      <ExploreButton />
    </main>
  );
}

export default Home;
