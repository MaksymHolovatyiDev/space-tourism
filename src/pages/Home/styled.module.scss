.root {
  height: 100vh;
  width: 100vw;

  display: flex;
  justify-content: space-between;
  align-items: flex-end;

  padding: 0 165px 131px;

  background-size: cover;
  background-repeat: no-repeat;
  background-image: url('./assets/background-home-desktop.jpg');

  @media screen and (max-width: 1300px) {
    flex-direction: column;
    align-items: center;

    padding: 96px 159px 90px;

    background-image: url('./assets/background-home-tablet.jpg');
  }

  @media screen and (max-width: 615px) {
    background-image: url('./assets/background-home-mobile.jpg');

    padding: 112px 24px 48px;
  }
}

.container {
  width: 450px;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;

  gap: 24px;

  @media screen and (max-width: 1300px) {
    margin-top: 106px;

    align-items: center;
  }

  @media screen and (max-width: 615px) {
    width: auto;

    text-align: center;

    margin: 0;
  }
}
