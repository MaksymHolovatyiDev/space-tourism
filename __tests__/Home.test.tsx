import renderer from 'react-test-renderer';
import Home from '../src/pages/Home';
import React from 'react';

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
