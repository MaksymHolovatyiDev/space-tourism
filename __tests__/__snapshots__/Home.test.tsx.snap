// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <div>
    <p
      className="undefined undefined"
    >
      So, you wat to travel to
    </p>
    <h1
      className="undefined undefined undefined"
    >
      Space
    </h1>
    <p
      className="undefined undefined"
    >
      Let’s face it; if you want to go to space, you might as well genuinely go to outer space and not hover kind of on the edge of it. Well sit back, and relax because we’ll give you a truly out of this world experience!
    </p>
  </div>
  <button
    type="button"
  >
    Explore
  </button>
</main>
`;
